go/store/nbs/cmp_chunk_table_writer.go

```diff
@@ -80,11 +75,6 @@ func (tw *CmpChunkTableWriter) AddCmpChunk(c CompressedChunk) error {
 		panic("NBS blocks cannot be zero length")
 	}
 
-	if tw.chunkHashes.Has(c.H) {
```

Comment:
Removing this check changes the contract for the interface. Now the caller has to ensure no duplicate chunks or you end up with duplicates in the table files produced.

Is that expectation documented anywhere? Not super familiar with this set of files, but it worries me a bit that we aren't consistent with how these interfaces are expected to be used

Reply:
It does change the contract, but in the past, you would get an error --- not produce a table file that had duplicate chunks.

It's perfectly valid for a table file to have duplicate chunks, and we create them that way by design when we conjoin table files. I don't know why this check exists, but I'm guessing it was here to check prevent regressions where upstream code changed in a way that created a bunch of duplicate chunks in a written table file.

If we wanted to retain close-to-them-same semantics, we could preform the uniqueness check when we write the index. We could also do that, but optionally enable it and use that when running bats and unit tests but not in the shipped product.

Definitely this hash map isn't carrying its weight, but I'm not even sure the check itself is carrying its weight...